VHbbAnalysis/Heppy/python/AdditionalBoost.py

```diff
@@ -1116,6 +1117,28 @@ def process(self, event):
                 jet.JEC_L2L3Unc = tmp_jet.jetEnergyCorrUncertainty
 		jet.JEC_L1L2L3 = self.jetReCalibratorAK8L1L2L3.getCorrection(tmp_jet,rho)
                 jet.JEC_L1L2L3Unc = tmp_jet.jetEnergyCorrUncertainty
+		jet.puppi_pt             = jet.userFloat("ak8PFJetsPuppiValueMap:pt")
+                jet.puppi_mass       = jet.userFloat("ak8PFJetsPuppiValueMap:mass")
+                jet.puppi_eta           = jet.userFloat("ak8PFJetsPuppiValueMap:eta")
+                jet.puppi_phi           = jet.userFloat("ak8PFJetsPuppiValueMap:phi")
+                jet.puppi_tau1         = jet.userFloat("ak8PFJetsPuppiValueMap:NjettinessAK8PuppiTau1")
+	        jet.puppi_tau2         = jet.userFloat("ak8PFJetsPuppiValueMap:NjettinessAK8PuppiTau2")
+
+                puppi_softdrop= ROOT.TLorentzVector()
+		puppi_softdrop_raw= ROOT.TLorentzVector()
+                puppi_softdrop_subjet_raw=ROOT.TLorentzVector()	
+                puppi_softdrop_subjet=ROOT.TLorentzVector()
+                sbSubjetsPuppi = jet.subjets("SoftDropPuppi")
+                for it in sbSubjetsPuppi  :
+                       puppi_softdrop_subjet.SetPtEtaPhiM(it.pt(),it.eta(),it.phi(),it.mass())
+                       puppi_softdrop+=puppi_softdrop_subjet
+		       puppi_softdrop_subjet_raw.SetPtEtaPhiM(it.correctedP4(0).pt(),it.correctedP4(0).eta(),it.correctedP4(0).phi(),it.correctedP4(0).mass())
+                       puppi_softdrop_raw+=puppi_softdrop_subjet_raw
+
+                jet.puppi_msoftdrop       = puppi_softdrop.M()
+		jet.puppi_msoftdrop_raw   = puppi_softdrop_raw.M()
+                jet.puppi_msoftdrop_corrL2L3       = puppi_softdrop.M()*corr
```

Comment:
yes, but anyway that it's not used. Dedicated corrections are applied later starting from the raw mass.